tests/devices/test_default_clifford.py

```diff
@@ -0,0 +1,427 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the tests for the clifford simulator based on stim
+"""
+import os
+import sys
+import pytest
+import numpy as np
+import pennylane as qml
+
+from pennylane.devices.default_clifford import _import_stim
+
+stim = pytest.importorskip("stim")
+
+pytestmark = pytest.mark.external
+
+
+def circuit_1():
+    """Circuit 1 with Clifford gates."""
+    qml.GlobalPhase(np.pi)
+    qml.CNOT(wires=[0, 1])
+    qml.PauliX(wires=[1])
+    qml.Barrier()
+    qml.ISWAP(wires=[0, 1])
+    qml.Hadamard(wires=[0])
+
+
+@pytest.mark.parametrize("circuit", [circuit_1])
+@pytest.mark.parametrize(
+    "expec_op",
+    [
+        qml.PauliZ(0),
+        qml.PauliZ(0) @ qml.PauliX(1),
+        qml.Hamiltonian([0.42], [qml.PauliZ(0) @ qml.PauliX(1)]),
+    ],
+)
+def test_expectation_clifford(circuit, expec_op):
+    """Test that execution of default.clifford is possible and agrees with default.qubit."""
+    dev_c = qml.device("default.clifford")
+    dev_q = qml.device("default.qubit")
+
+    def circuit_fn():
+        circuit()
+        return qml.expval(expec_op)
+
+    qnode_clfrd = qml.QNode(circuit_fn, dev_c)
+    qnode_qubit = qml.QNode(circuit_fn, dev_q)
+    assert np.allclose(qnode_clfrd(), qnode_qubit())
+
+
+@pytest.mark.parametrize("circuit", [circuit_1])
+def test_grad_clifford(circuit):
+    """Test that gradients of default.clifford agrees with default.qubit."""
+    dev_c = qml.device("default.clifford")
+    dev_q = qml.device("default.qubit")
+    assert dev_c.supports_derivatives()
+    assert not dev_c.supports_jvp()
+    assert not dev_c.supports_vjp()
+
+    def circuit_fn():
+        circuit()
+        return qml.expval(qml.PauliZ(0))
+
+    qnode_clfrd = qml.QNode(circuit_fn, dev_c)
+    qnode_qubit = qml.QNode(circuit_fn, dev_q)
+    res_q, grad_q = qnode_qubit(), qml.grad(qnode_qubit)()
```

Comment:
I'm confused, how are we doing gradients if the circuit is not parametrized?

Reply:
Since the circuits are not parameterized, the gradients returned from this device will be zero.

Reply:
Removed it.